project/manager_test.go

```diff
@@ -1,41 +1,195 @@
 package project
 
 import (
+	"github.com/machinefi/sprout/persistence/contract"
+	"os"
 	"testing"
 
 	"github.com/agiledragon/gomonkey/v2"
 	"github.com/ethereum/go-ethereum/ethclient"
 	"github.com/ethereum/go-ethereum/rpc"
-	"github.com/machinefi/sprout/smartcontracts/go/project"
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/require"
+
+	"github.com/machinefi/sprout/smartcontracts/go/project"
 )
 
 func TestNewManager(t *testing.T) {
 	r := require.New(t)
-	p := gomonkey.NewPatches()
-	defer p.Reset()
 
 	t.Run("FailedToDialChain", func(t *testing.T) {
+		p := gomonkey.NewPatches()
+		defer p.Reset()
+
 		p = p.ApplyFuncReturn(ethclient.Dial, nil, errors.New(t.Name()))
 
 		_, err := NewManager("", "", "", "", "")
 		r.ErrorContains(err, t.Name())
 	})
-	p = p.ApplyFuncReturn(ethclient.Dial, ethclient.NewClient(&rpc.Client{}), nil)
 
 	t.Run("FailedToNewContracts", func(t *testing.T) {
+		p := gomonkey.NewPatches()
+		defer p.Reset()
+
+		p = p.ApplyFuncReturn(ethclient.Dial, ethclient.NewClient(&rpc.Client{}), nil)
```

Comment:
p.ApplyFuncReturn(ethclient.Dial, ethclient.NewClient(&rpc.Client{}), nil)
is ok, 
don't need assignment to p again, for it's p's pointer